csharp/src/Ice/Protocol.cs

```diff
@@ -53,14 +53,25 @@ internal static void CheckSupported(this Protocol protocol)
         /// <returns>The parsed protocol, or throws an exception if the string cannot be parsed.</returns>
         internal static Protocol Parse(string str)
         {
-            return str switch
+            switch (str)
             {
-                "ice1" => Protocol.Ice1,
-                "ice2" => Protocol.Ice2,
-                // Parse as a Major.Minor encoding:
-                _ => Encoding.TryParse(str, out Encoding encoding) && encoding.Minor == 0 ?
-                    (Protocol)encoding.Major : (Protocol)byte.Parse(str, CultureInfo.InvariantCulture)
-            };
+                case "ice1":
+                    return Protocol.Ice1;
+                case "ice2":
+                    return Protocol.Ice2;
+                default:
+                    // TODO: for now, parse as a Major.Minor encoding or just Major:
+                    byte b = Encoding.TryParse(str, out Encoding encoding) && encoding.Minor == 0 ?
```

Comment:
I don't think reusing Encoding.TryParse for Protocol is the best, byte.Parse here will throw a generic FormatException about parsing a byte, I think will be clear to implement this without rely on Encoding.TryParse

Reply:
Yes, this was introduced by a previous PR. This code will be reworked with the URI parsing.